### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by CultureBunkerRecordings**

### CODE REVIEW

File: Highway.cpp [hunk 1]
```diff
@@ -18,22 +18,47 @@ void Highway::changeSpeed(int newSpeed)
...
+        vehicle->closeWindows();
+    }
+    if (auto* vehicle = dynamic_cast<Motorcycle*>(v))
+    {
+        vehicle->setSpeed(60);
+    }
+    if (auto* vehicle = dynamic_cast<SemiTruck*>(v))
+    {
+        vehicle->stop();
+    }
```
Comment (matkatmusic) [inline Highway.cpp]: - [ ] use `else if`.  if one cast succeeds you should not continue trying to cast.
- [ ] Motorcycle: Call the other member function that doesn't evade the cops.
- [ ] You need to write a `SemiTruck` member function that does something other than `setSpeed(0)`.  What you're calling here gets called in `removeVehicleInternal()`.  See instruction 3 in main.cpp

File: HighwayPatrol.cpp [hunk 2]
```diff
@@ -6,11 +6,16 @@ HighwayPatrol::HighwayPatrol() : Vehicle("HighwayPatrol")
 
 }
 
+HighwayPatrol::~HighwayPatrol()
+{
+    
+}
+
```
Comment (matkatmusic) [inline HighwayPatrol.cpp]: you can `= default;` this instead of providing an empty implementation.

File: HighwayPatrol.cpp [hunk 3]
```diff
@@ -28,7 +33,7 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     if( willArrest )
     {
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " << v->name<< " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
```
Comment (matkatmusic) [inline HighwayPatrol.cpp]: `v->name` is not the vehicle type.
- [ ] revise according to spec.

File: SemiTruck.h [hunk 4]
```diff
@@ -3,4 +3,10 @@
 
 struct SemiTruck : public Vehicle
 {
+    SemiTruck(const std::string s);
+    virtual ~SemiTruck();
+    SemiTruck(const SemiTruck&) = default;
+    SemiTruck& operator=(const SemiTruck&) = default;
+
+    void stop();
```
Comment (matkatmusic) [inline SemiTruck.h]: - [ ] Add another member function for use when `SemiTruck`'s enter the highway.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: HighwayPatrol.cpp (hunk 2 above)
Comment (CultureBunkerRecordings) [reply HighwayPatrol.cpp]: would not work with =default in header file got a vtable warning, no out of line virtual methods..

File: HighwayPatrol.cpp (hunk 3 above)
Comment (CultureBunkerRecordings) [reply HighwayPatrol.cpp]: need a bit of help here not sure how to print the actual object type.. car, motorcycle, semitruck etc..

**Review by matkatmusic — APPROVED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**